src/parser/ParseNode.mts

```diff
@@ -0,0 +1,2328 @@
+    readonly B: BitwiseXORExpressionOrHigher;
+  }
+
+  // LogicalANDExpression :
+  //   BitwiseORExpression
+  //   LogicalANDExpression `&&` BitwiseORExpression
+  export type LogicalANDExpressionOrHigher =
+    | BitwiseORExpressionOrHigher
+    | LogicalANDExpression;
+
+  // LogicalANDExpression (partial) :
+  //   LogicalANDExpression `&&` BitwiseORExpression
+  export interface LogicalANDExpression extends BaseParseNode {
+    readonly type: 'LogicalANDExpression';
+    readonly LogicalANDExpression: LogicalANDExpressionOrHigher;
+    readonly BitwiseORExpression: BitwiseORExpressionOrHigher;
+  }
+
+  // LogicalORExpression :
+  //   LogicalANDExpression
+  //   LogicalORExpression `||` LogicalANDExpression
+  export type LogicalORExpressionOrHigher =
+    | LogicalANDExpressionOrHigher
+    | LogicalORExpression;
+
+  // LogicalORExpression (partial) :
+  //   LogicalORExpression `||` LogicalANDExpression
+  export interface LogicalORExpression extends BaseParseNode {
+    readonly type: 'LogicalORExpression';
+    readonly LogicalORExpression: LogicalORExpressionOrHigher;
+    readonly LogicalANDExpression: LogicalANDExpressionOrHigher;
+  }
+
+  // CoalesceExpression :
+  //   CoalesceExpressionHead `??` BitwiseORExpression
+  export interface CoalesceExpression extends BaseParseNode {
+    readonly type: 'CoalesceExpression';
+    readonly CoalesceExpressionHead: CoalesceExpressionHead;
+    readonly BitwiseORExpression: BitwiseORExpressionOrHigher;
+  }
+
+  // CoalesceExpressionHead :
+  //   CoalesceExpression
+  //   BitwiseORExpression
+  export type CoalesceExpressionHead =
+    | BitwiseORExpressionOrHigher
+    | CoalesceExpression;
+
+  // ShortCircuitExpression :
+  //   LogicalORExpression
+  //   CoalesceExpression
+  export type ShortCircuitExpressionOrHigher =
+    | LogicalORExpressionOrHigher
+    | CoalesceExpression;
+
+  // ConditionalExpression :
+  //   ShortCircuitExpression
+  //   ShortCircuitExpression `?` AssignmentExpression `:` AssignmentExpression
+  export type ConditionalExpressionOrHigher =
+    | ShortCircuitExpressionOrHigher
+    | ConditionalExpression;
+
+  // ConditionalExpression (partial) :
+  //   ShortCircuitExpression `?` AssignmentExpression `:` AssignmentExpression
+  export interface ConditionalExpression extends BaseParseNode {
+    readonly type: 'ConditionalExpression';
+    readonly ShortCircuitExpression: ShortCircuitExpressionOrHigher;
+    readonly AssignmentExpression_a: AssignmentExpressionOrHigher;
+    readonly AssignmentExpression_b: AssignmentExpressionOrHigher;
+  }
+
+  // AssignmentExpression :
+  //   ConditionalExpression
+  //   [+Yield] YieldExpression
+  //   ArrowFunction
+  //   AsyncArrowFunction
+  //   LeftHandSideExpression `=` AssignmentExpression
+  //   LeftHandSideExpression AssignmentOperator AssignmentExpression
+  //   LeftHandSideExpression LogicalAssignmentOperator AssignmentExpression
+  export type AssignmentExpressionOrHigher =
```

Comment:
I wonder what this `OrHigher` exactly means, I know it is related to the syntax "hierarchy". I saw it in the TypeScript compiler too.

Reply:
It means "this node, or a node of a higher precedence".